webhook/server/server.go

```diff
@@ -0,0 +1,437 @@
+
+	validNameRegex, _ := regexp.Compile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
+	for _, unit := range []string{namespace, name, netInterface} {
+		ok := validNameRegex.MatchString(unit)
+		if !ok && len(unit) > 0 {
+			return networkSelectionElement, logging.Errorf("at least one of the network selection units is invalid: error found at '%s'", unit)
+		}
+	}
+
+	networkSelectionElement = &types.NetworkSelectionElement{
+		Namespace:        namespace,
+		Name:             name,
+		InterfaceRequest: netInterface,
+	}
+
+	return networkSelectionElement, nil
+}
+
+func parsePodNetworkSelections(podNetworks string) ([]*types.NetworkSelectionElement, error) {
+	var networkSelections []*types.NetworkSelectionElement
+
+	if len(podNetworks) == 0 {
+		return nil, logging.Errorf("empty string passed as network selection elements list")
+	}
+
+	/* try to parse as JSON array */
+	err := json.Unmarshal([]byte(podNetworks), &networkSelections)
+
+	/* if failed, try to parse as comma separated */
+	if err != nil {
+		logging.Debugf("%s is not in JSON format: %s. Will try to parse as comma separated network selections list", podNetworks, err.Error())
+		for _, networkSelection := range strings.Split(podNetworks, ",") {
+			networkSelection = strings.TrimSpace(networkSelection)
+			networkSelectionElement, err := parsePodNetworkSelectionElement(networkSelection)
+			if err != nil {
+				return nil, logging.Errorf("error parsing network selection element: %v", err)
+			}
+			networkSelections = append(networkSelections, networkSelectionElement)
+		}
+	}
+
+	/* fill missing namespaces with default value */
+	for _, networkSelection := range networkSelections {
+		if networkSelection.Namespace == "" {
+			networkSelection.Namespace = "default"
+		}
+	}
+
+	return networkSelections, nil
+}
+
+func mutateHandler(w http.ResponseWriter, req *http.Request) {
+	logging.Debugf("Received mutation request")
+
+	/* read AdmissionReview from the HTTP request */
+	ar, httpStatus, err := readAdmissionReview(req)
+	if err != nil {
+		http.Error(w, err.Error(), httpStatus)
+		return
+	}
+
+	/* read pod annotations */
+	/* if networks missing skip everything */
+	pod, err := deserializePod(ar)
+	if err != nil {
+		handleValidationError(w, ar, err)
+		return
+	}
+	if netSelections, exists := pod.ObjectMeta.Annotations[networksAnnotationKey]; exists && netSelections != "" {
+		/* map of resources request needed by a pod and a number of them */
+		resourceRequests := make(map[string]int64)
+
+		/* unmarshal list of network selection objects */
+		networks, _ := parsePodNetworkSelections(netSelections)
+
+		for _, n := range networks {
+			/* for each network in annotation ask API server for network-attachment-definition */
+			networkAttachmentDefinition, err := getNetworkAttachmentDefinition(n.Namespace, n.Name)
+			if err != nil {
+				/* if doesn't exist: deny pod */
+				explanation := logging.Errorf("could not find network attachment definition '%s/%s': %s", n.Name, n.Namespace, err)
```

Comment:
I think the order of name and namespace in the error message should be swapped to reflect how we encode it in networks annotation (namespace/networkName).

Reply:
done